### PR TITLE
You can no longer create non-synthesizable chemicals with bees.

### PR DESCRIPTION
**Pull request by Iamgoofball**
## About The Pull Request

You can no longer create non-synthesizable chemicals with bees.

## Why It's Good For The Game

Fixes the ability to mass produce reagents that aren't supposed to be directly synthesized.

## Changelog

:cl:
fix: You can no longer create non-synthesizable chemicals with bees.
/:cl:

### CODE REVIEW

**Comment by cnleth** [comment]
why exactly is a bee "fearful of the non-synthesizable contents!"? Really lazy explanation text

**Comment by mikederkan** [comment]
why is this good for the game, exactly? it just seems like removal of funny gimmick potential

**Comment by Exester509** [comment]
Will this affect all bees or only crew-obtainable ones? The wizard's summon bees spell does benefit from the presence of these chemicals, and it's not like the crew can endlessly abuse those

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 1]
```diff
@@ -258,6 +258,9 @@
 	var/datum/reagent/chemical = needle.reagents.get_master_reagent()
 	if(isnull(chemical))
 		return
+	if(!(chemical.chemical_flags & REAGENT_CAN_BE_SYNTHESIZED))
+		to_chat(user, span_warning("[queen] backs away from the syringe, fearful of the non-synthesizable contents!"))
```
Comment (ZephyrTFA) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm:262]:
```suggestion
		to_chat(user, span_warning("[queen] does not react to the contents of the syringe. It appears that the contents are not able to be synthesized naturally!"))
```

**Comment by Serxule** [comment]
> Will this affect all bees or only crew-obtainable ones? The wizard's summon bees spell does benefit from the presence of these chemicals, and it's not like the crew can endlessly abuse those

bee-nades, holodeck, wizard bee spell, all of those rely on bees producing random chems (a lot of which are likely not synthesizeable)

**Comment by ZephyrTFA** [comment]
> > Will this affect all bees or only crew-obtainable ones? The wizard's summon bees spell does benefit from the presence of these chemicals, and it's not like the crew can endlessly abuse those
> 
> bee-nades, holodeck, wizard bee spell, all of those rely on bees producing random chems (a lot of which are likely not synthesizeable)

This only changes adding chemicals to the bees via syringe. Please read the PR before making comments.

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 2]
```diff
@@ -258,6 +258,9 @@
 	var/datum/reagent/chemical = needle.reagents.get_master_reagent()
 	if(isnull(chemical))
 		return
+	if(!(chemical.chemical_flags & REAGENT_CAN_BE_SYNTHESIZED))
+		to_chat(user, span_warning("[queen] does not react to the contents of the syringe. It appears that the contents are not able to be synthesized naturally!"))
+		return
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm:263]: the text is a bit weird considering you dont actually inject the syringe yet.

Should be something like 

` "[chemical.name] cannot be inserted into a bee's genome."`

Comment (Iamgoofball) [reply code/modules/mob/living/basic/farm_animals/bee/_bee.dm:263]: hash this out with @ZephyrTFA

Comment (ZephyrTFA) [reply code/modules/mob/living/basic/farm_animals/bee/_bee.dm:263]: shit talking my code recommendation, smh smh

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 3]
```diff
@@ -258,6 +258,9 @@
 	var/datum/reagent/chemical = needle.reagents.get_master_reagent()
 	if(isnull(chemical))
 		return
+	if(!(chemical.chemical_flags & REAGENT_CAN_BE_SYNTHESIZED))
+		to_chat(user, span_warning("[queen] does not react to the contents of the syringe. It appears that the contents are not able to be synthesized naturally!"))
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm:262]:
```suggestion
		to_chat(user, span_warning("[chemical.name] cannot be inserted into a bee's genome!"))
```

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm (hunk 2 above)
Comment (ZephyrTFA) [reply code/modules/mob/living/basic/farm_animals/bee/_bee.dm:263]:
```suggestion
	if(!(chemical.chemical_flags & REAGENT_CAN_BE_SYNTHESIZED))
		to_chat(user, span_warning("[chemical.name] cannot be inserted into a bee's genome."))
		return
```

**Comment by Serxule** [comment]
> This only changes adding chemicals to the bees via syringe. Please read the PR before making comments.

i did read the PR, it says nothing about that

**Comment by Fikou** [comment]
i thought this was already a thing

**Comment by MrMelbert** [comment]
#56995

**Comment by Exester509** [comment]
> > This only changes adding chemicals to the bees via syringe. Please read the PR before making comments.
> 
> i did read the PR, it says nothing about that

I just checked the code and this seemingly just affects injections, so that shouldn't affect poison bees

**Comment by bluetshirtguy** [comment]
Bee's have been broken for ages and we're removing some of the funsies from them anyways IM CRYING

**Comment by ZeroKelvn** [comment]
Soft disagree, bees are still an incredibly rare occurrence to see (on Manuel at least), and even rarer to see them mixed with another chem. Besides, to inject them with a nonsynth chem, you'd still need to get your hands on a dose to start it off in the first place.

Completely unrelated but this also made me realize the potential of Epinephrine/Atropine and Penthrite beehives.

**Comment by moocowswag** [comment]
> > > This only changes adding chemicals to the bees via syringe. Please read the PR before making comments.
> > 
> > 
> > i did read the PR, it says nothing about that
> 
> I just checked the code and this seemingly just affects injections, so that shouldn't affect poison bees

Then its completly relevant because we have even less consistency with bees now, if some bees can sting synthesizeable chems and some cant.

**Pull request merged**